### PR TITLE
Remove fetch content

### PR DESCRIPTION
**Pull request by JasonMarechal25**
See ADR for more information
- Remove fetch_content usage for runtime dependencies
- Create an action to install minizip during CI

### CODE REVIEW

**Review by flomnes — APPROVED** [review]

**Review by flomnes — CHANGES_REQUESTED** [review]
Find an alternative for developers. How many command-lines required to build from scratch ?

**Comment by JasonMarechal25** [comment]
> Find an alternative for developers. How many command-lines required to build from scratch ?

We can't do much more. Dependencies need to be installed. Fetch_content is not the way to manage runtime dependencies. CMake is build system, not a package manager, that's a job for vcpkg or conan.

Most dependencies are installed thanks to them being available in apt or vcpkg. Ortools isn't, minizip may be present in vcpkg. We could always provide scripts to install them, but it is generally expected from developers to know how to build and install third partie and from maintainer to disclose dependencies.

File: .github/workflows/windows-vcpkg.yml [hunk 1]
```diff
@@ -132,6 +132,8 @@ jobs:
     - name: Install pip dependencies if necessary
       run: pip install -r src/tests/examples/requirements.txt
 
+    - uses: ./.github/workflows/install_minizip
+
```
Comment (flomnes) [inline .github/workflows/windows-vcpkg.yml:136]: Use vcpkg to install minizip-ng ? It is available
image

File: src/packaging/Config.cmake.in [hunk 2]
```diff
@@ -1,8 +1,10 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
-find_dependency(ortools)
+
+#In this order. Why? I don't really know, but it works.
 find_dependency(minizip)
+find_dependency(ortools)
```
Comment (flomnes) [inline src/packaging/Config.cmake.in:7]: zlib shenanigans, probably

**Review by flomnes — COMMENTED** [review]
We should try to use vcpkg as our dependency manager to ease our work locally & on GH actions

**Pull request closed without merging**